nautobot/virtualization/api/views.py

```diff
@@ -93,8 +94,34 @@ def get_serializer_class(self):
         return serializers.VirtualMachineWithConfigContextSerializer
 
 
-class VMInterfaceViewSet(ModelViewSet):
-    queryset = VMInterface.objects.prefetch_related("virtual_machine", "tags", "tagged_vlans")
-    serializer_class = serializers.VMInterfaceSerializer
+@extend_schema_view(
+    bulk_update=extend_schema(responses={"200": serializers.VMInterfaceSerializer(many=True)}, versions=["1.2"]),
+    bulk_partial_update=extend_schema(
+        responses={"200": serializers.VMInterfaceSerializer(many=True)}, versions=["1.2"]
+    ),
+    create=extend_schema(responses={"201": serializers.VMInterfaceSerializer}, versions=["1.2"]),
+    list=extend_schema(responses={"200": serializers.VMInterfaceSerializer(many=True)}, versions=["1.2"]),
+    partial_update=extend_schema(responses={"200": serializers.VMInterfaceSerializer}, versions=["1.2"]),
+    retrieve=extend_schema(responses={"200": serializers.VMInterfaceSerializer}, versions=["1.2"]),
+    update=extend_schema(responses={"200": serializers.VMInterfaceSerializer}, versions=["1.2"]),
+)
+class VMInterfaceViewSet(StatusViewSetMixin, ModelViewSet):
+    queryset = VMInterface.objects.prefetch_related("virtual_machine", "status", "tags", "tagged_vlans")
+    serializer_class = serializers.VMInterfaceSerializerVersion13
     filterset_class = filters.VMInterfaceFilterSet
     brief_prefetch_fields = ["virtual_machine"]
+
+    def get_serializer_class(self):
+        if (
+            not self.brief
+            and not getattr(self, "swagger_fake_view", False)
+            and (
+                not hasattr(self.request, "major_version")
+                or self.request.major_version > 1
+                or (self.request.major_version == 1 and self.request.minor_version < 3)
+            )
+        ):
```

Comment:
We're probably reaching a point where it would be desirable to add a helper method on `ModelViewSet` that could encapsulate this kind of repeated logic.